src/NovelLoggingService.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#include <iostream>
+#include "NovelLoggingService.h"
+#include "NovelRTUtilities.h"
+#include "spdlog/async.h"
+
+namespace NovelRT {
+
+NovelLoggingService::NovelLoggingService() {
+  try
+  {
+    _console = spdlog::get(NovelUtilities::CONSOLE_LOG_GENERIC);
+    if (_console == nullptr)
+    {
+        _console = spdlog::stdout_color_mt<spdlog::async_factory>(NovelUtilities::CONSOLE_LOG_GENERIC);
+    }
+
+    #ifndef NDEBUG
+    setLogLevel(LogLevel::DEBUG);
+    #else
+    setLogLevel(LogLevel::INFO);
+    #endif
+
+    //Set spdlog's error handler in case it fails.
+     spdlog::set_error_handler([](const std::string& msg) {
+        std::cerr << "SPDLOG ERROR: " << msg << std::endl;
+    });
+  }
+  catch (const spdlog::spdlog_ex &ex)
+  {
+    std::cout << "Log System failed to initialize: " << ex.what() << std::endl;
+  }
+}
+
+NovelLoggingService::NovelLoggingService(std::string core) {
```

Comment:
Shouldn't `std::string` be passed by reference or is that just me being silly?

Reply:
Learning op. for me - fixed just like the last message was.